src/transformers/models/gpt2/modeling_gpt2.py

```diff
@@ -394,6 +431,17 @@ def _init_weights(self, module):
             module.bias.data.zero_()
             module.weight.data.fill_(1.0)
 
+        # Reinitialize selected weights subject to the OpenAI GPT-2 Paper Scheme:
+        #   > A modified initialization which accounts for the accumulation on the residual path with model depth. Scale
+        #   > the weights of residual layers at initialization by a factor of 1/√N where N is the # of residual layers.
+        #   >   -- GPT-2 :: https://openai.com/blog/better-language-models/
+        #
+        # Reference (Megatron-LM): https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/model/gpt_model.py
+        for name, p in module.named_parameters():
```

Comment:
As I understood it the `c_proj` initialization was then never correctly initialized as advertised by the GPT2 paper and this is more or less a bug fix, no?

Reply:
Yeah, I believe so!

Reply:
@patrickvonplaten, shouldn't this code go into `def _init_weights(self, module)` - otherwise we will be always initializing, even when loading from pretrained?